### PR TITLE
docker-slim 1.18 (new formula)

### PR DESCRIPTION
**Pull request by sgerrand**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
This is a formula for [Docker Slim](https://github.com/docker-slim/docker-slim), a tool for minifying and securing Docker containers.

I've added a fairly minimal test case, mainly because I was loathe to add a dependency on Docker in order to run tests.

### CODE REVIEW

**Comment by JCount** [comment]
@BrewTestBot test this please

**Review by ilovezfs — DISMISSED** [review]
In new formulae, we generally want to avoid `go_resource` blocks. Please file an issue upstream requesting use of a Go vendor directory and/or one of the various Go vendoring tools.

**Comment by sgerrand** [comment]
> In new formulae, we generally want to avoid `go_resource` blocks. Please file an issue upstream requesting use of a Go vendor directory and/or one of the various Go vendoring tools.

The dependencies were vendorized in v1.18. I've updated this formula to reference that version and changed the way it builds based on your comment.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

File: Formula/docker-slim.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+   --log value            log file to store logs
+   --log-format value     set the format used by logs ('text' (default), or 'json') (default: "text")
+   --tls                  use TLS
+   --tls-verify           verify TLS
+   --tls-cert-path value  path to TLS cert files
+   --host value           Docker host address
+   --help, -h             show help
+   --version, -v          print the version
+EOUSAGE
+    assert_match usage, shell_output("#{bin}/docker-slim --help")
```
Comment (ilovezfs) [inline Formula/docker-slim.rb]: This needs to be a test that exercises and verifies the functionality of the software, not the help output.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
```
==> Checking out tag 1.18
git checkout -f 1.18 --
HEAD is now at 8c2da78... restructure
git reset --hard 1.18
HEAD is now at 8c2da78 restructure
==> go build -v ./cmd/docker-slim
cmd/docker-slim/cli.go:13:2: cannot find package "github.com/Sirupsen/logrus" in any of:
	/usr/local/Cellar/go/1.9.1/libexec/src/github.com/Sirupsen/logrus (from $GOROOT)
	/private/tmp/docker-slim-20171007-70978-i2afva/src/github.com/Sirupsen/logrus (from $GOPATH)
cmd/docker-slim/opts.go:13:2: cannot find package "github.com/cloudimmunity/go-dockerclientx" in any of:
	/usr/local/Cellar/go/1.9.1/libexec/src/github.com/cloudimmunity/go-dockerclientx (from $GOROOT)
	/private/tmp/docker-slim-20171007-70978-i2afva/src/github.com/cloudimmunity/go-dockerclientx (from $GOPATH)
cmd/docker-slim/cli.go:14:2: cannot find package "github.com/codegangsta/cli" in any of:
	/usr/local/Cellar/go/1.9.1/libexec/src/github.com/codegangsta/cli (from $GOROOT)
	/private/tmp/docker-slim-20171007-70978-i2afva/src/github.com/codegangsta/cli (from $GOPATH)
cmd/docker-slim/opts.go:14:2: cannot find package "github.com/docker/go-connections/nat" in any of:
	/usr/local/Cellar/go/1.9.1/libexec/src/github.com/docker/go-connections/nat (from $GOROOT)
	/private/tmp/docker-slim-20171007-70978-i2afva/src/github.com/docker/go-connections/nat (from $GOPATH)
```
@sgerrand maybe there are missing items from vendor directory?

**Pull request closed without merging**